### PR TITLE
feat(appeals-service-app): appeal statement business validation

### PR DESCRIPTION
**Pull request by frank-dodion**
## Ticket Number
UCD-1213

## Description of change
implemented business logic validation for appeal statement file upload

### CODE REVIEW

**Review by rcmsoftwaredevelopment — APPROVED** [review]

File: appeals-service-api/src/__tests__/appeals.spec.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const { appealDocument } = require('../models/appeal');
 jest.mock('../db/db');
 
 async function createAppeal() {
-  const appeal = appealDocument;
+  const appeal = JSON.parse(JSON.stringify(appealDocument));
```
Comment (MrSimonEmms) [inline appeals-service-api/src/__tests__/appeals.spec.js:11]: What's the story here? Surely this is the same as `appealDocument`

File: appeals-service-api/src/__tests__/appeals.spec.js [hunk 2]
```diff
@@ -34,11 +34,8 @@ describe('Appeals API', () => {
   });
 
   test('POST /api/v1/appeals - It responds with a newly created appeal', async () => {
-    const appeal = appealDocument;
-    const response = await request(app).post('/api/v1/appeals').send({
-      // lpaCode: 'E60000281',
-      // decisionDate,
-    });
+    const appeal = JSON.parse(JSON.stringify(appealDocument));
```
Comment (MrSimonEmms) [inline appeals-service-api/src/__tests__/appeals.spec.js:37]: As above

File: appeals-service-api/src/controllers/appeals.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const { validateAppeal } = require('../middleware/validateAppeal');
 
 module.exports = {
   async create(req, res) {
-    const appeal = appealDocument;
+    const appeal = JSON.parse(JSON.stringify(appealDocument));
```
Comment (MrSimonEmms) [inline appeals-service-api/src/controllers/appeals.js:10]: +1

**Review by MrSimonEmms — COMMENTED** [review]
Confused by the `JSON.parse(JSON.stringify(obj))`

File: appeals-service-api/src/__tests__/appeals.spec.js (hunk 1 above)
Comment (frank-dodion) [reply appeals-service-api/src/__tests__/appeals.spec.js:11]: I think we need a deep copy to avoid issues when tests etc are run in parallel.

**Review by MrSimonEmms — APPROVED** [review]
Ok. May be better ways of doing it, but can revisit it in the future (we probably never will)

**Pull request merged**